# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.cpp
# Commit: 5be8557c4721
# Full Hash: 5be8557c4721b4d1a2ac07bf5ae0516d9f2da2bc
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 09:10:34
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 11. Plumb in OffscreenCanvasDisplayHelper. r=jgilbert
#   
#   This patch integrates OffscreenCanvasDisplayHelper with
#   HTMLCanvasElement, OffscreenCanvas and nsDisplayCanvas to allow
#   asynchronous display of an OffscreenCanvas.
# ==============================================================================

diff -r 49f8b4205a46 -r 5be8557c4721 dom/canvas/WebGLContext.cpp
--- a/dom/canvas/WebGLContext.cpp	Thu Dec 09 19:25:28 2021 +0000
+++ b/dom/canvas/WebGLContext.cpp	Thu Dec 09 19:25:28 2021 +0000
@@ -228,12 +228,12 @@
   mIsCanvasDirty = true;
   mCapturedFrameInvalidated = true;
 
-  if (!mCanvasElement) {
-    return;
+  if (mCanvasElement) {
+    SVGObserverUtils::InvalidateDirectRenderingObservers(mCanvasElement);
+    mCanvasElement->InvalidateCanvasContent(nullptr);
+  } else if (mOffscreenCanvas) {
+    mOffscreenCanvas->QueueCommitToCompositor();
   }
-
-  SVGObserverUtils::InvalidateDirectRenderingObservers(mCanvasElement);
-  mCanvasElement->InvalidateCanvasContent(nullptr);
 }
 
 void WebGLContext::OnMemoryPressure() {